Configure Analytics and update examples and documentation

jgroh9 opened the pull request:
# What was done

- Update readme documentation
- Update documentation and examples based on the latest Couchbase documentation
- Add missing semi-colons to NodeJS file
- Add ability to configure the Analytics service
- Add ability to set the cluster name
- Add ability to set Eventing service RAM

brantburnett commented on README.md:
```diff
@@ -118,13 +123,35 @@ Within this file, you can define the `CREATE INDEX` statements for your bucket,
 
+### Analytics Dataset Setup
+
+To setup the analytics service datasets, add a directory under the `/startup/<bucketname>/analytics` folder. Within this folder create a text file named `dataset.n1ql`. For example, `/startup/default/analytics/dateset.n1ql`.  Note that the names are case sensitive.
```
```suggestion
To setup the analytics service datasets, add a directory under the `/startup/<bucketname>/analytics` folder. Within this folder create a text file named `dataset.n1ql`. For example, `/startup/default/analytics/dataset.n1ql`.  Note that the names are case sensitive.
```

brantburnett commented on README.md:
````diff
@@ -118,13 +123,35 @@ Within this file, you can define the `CREATE INDEX` statements for your bucket,
+
+Within this file, you can define the `CREATE DATASET` statements for your bucket, separated by semicolons. **IMPORTANT** Make sure to append the proper `USE` statement to each `CREATE DATASET` statement so that it's placed in the proper DATAVERSE. Additionaly, always end your file with a `CONNECT LINK Local;` statement.
+
+```sql
````
The documentation doesn't seem to align with the code/example, which appear to be using JSON files rather than N1QL files.

brantburnett commented on README.md:
```diff
@@ -37,17 +37,22 @@ COPY . /startup/
+
+Values for CB_SERVICES and CB_INDEXSTORAGE correspond to parameters for the [Couchbase REST API](https://docs.couchbase.com/server/current/rest-api/rest-node-provisioning.html).
+
+**NOTE:** If you configure `CB_SERVICES` to create the `analytics` service, make sure you set `CB_ANALYTICSRAM` to a minimum of `1024`.
```
```suggestion
**NOTE:** If you configure `CB_SERVICES` to create the `cbas` analytics service, make sure you set `CB_ANALYTICSRAM` to a minimum of `1024`.
```

brantburnett commented on startup/buckets.json:
```diff
@@ -2,11 +2,11 @@
     "name": "default",
     "ramQuotaMB": 100,
-    "bucketType": "membase",
+    "bucketType": "couchbase",
```
Did you test this change against our current minimum version of 5.0.1? It appears that "couchbase" was a supported value in 5.0.1 based on docs, and "membase" deprecated, but I want to make sure we've confirmed with a test.

jgroh9 replied:
I did not test this. I was going by the documentation for Couchbase 5.0 found here: https://docs.couchbase.com/server/5.0/cli/cbcli/couchbase-cli-bucket-create.html#options and here https://docs.couchbase.com/server/5.0/rest-api/rest-bucket-create.html#http-method-and-uri
Both of these pages show that `couchbase` is supported and also mention that `memcached` is supported indicating that `membase` isn't even the proper text to be using here.

brantburnett replied:
I've tested it, it's fine. The documentation does, in fact, match reality. I did find some other issues with old versions I need to look at, though, probably going to deprecate 5.0 and 5.1.

brantburnett reviewed (changes requested):
Thanks for the submission, I have a few suggestions and questions inline.

Also, it appears that the new support for RBAC users isn't covered in the readme, could you add that as well?

I also wouldn't complain if the work on RBAC users was separated from the work on analytics in a different PR, so it could be reviewed/merged independently. But don't worry about that if it's too much trouble.

jgroh9 commented:
I've addressed the comments and removed the RBAC capabilities in favor of creating a separate PR for that work. If you'd like me to test the `couchbase` bucketType change for 5.0.1 let me know.

brantburnett commented on example/sample/analytics/dataset.json:
```diff
@@ -0,0 +1,8 @@
+{
```
The `sample/docker-compose.yml` file needs `cbas` added to the list of services, otherwise, the presence of these files cause it to have errors.

brantburnett commented on README.md:
```diff
@@ -37,17 +37,22 @@ COPY . /startup/
+
+Values for CB_SERVICES and CB_INDEXSTORAGE correspond to parameters for the [Couchbase REST API](https://docs.couchbase.com/server/current/rest-api/rest-node-provisioning.html).
+
+**NOTE:** If you configure `CB_SERVICES` to create the `cbas` analytics service, make sure you set `CB_ANALYTICSRAM` to a minimum of `1024`.
```
It just occurred to me, perhaps we should put a default value for CB_ANALYTICSRAM and CB_EVENTINGRAM in the Dockerfile?

brantburnett reviewed (changes requested):
Thanks again, I've found a few more details in my testing and further review.

jgroh9 commented:
Requested changes have been made
